packages/nuxt3/src/app/composables/fetch.ts

```diff
@@ -50,15 +51,16 @@ export function useFetch<
     ]
   }
 
-  const asyncData = useAsyncData(key, () => {
-    return $fetch(_request.value, _fetchOptions) as Promise<_ResT>
+  const asyncData = useAsyncData<_ResT, ErrorT, Transform, PickKeys>(key, () => {
+    return $fetch(_request.value, _fetchOptions)
   }, _asyncDataOptions)
 
   return asyncData
 }
 
 export function useLazyFetch<
   ResT = void,
+  ErrorT = any,
```

Comment:
Should we default it to `Error`? 

```suggestion
  ErrorT = Error,
```

Reply:
It will be a boolean on client-side, so we could default it to `boolean | Error`:
```suggestion
  ErrorT = boolean | Error,
```

Reply:
In this case, maybe we should have something similar to `_ResT` maybe? So that even if user provides a custom error type, we include possibility of boolean value.

Reply:
I think we should probably allow for user to provide a custom error transformation to return a custom error so it's not _always_ boolean on client-side: https://github.com/nuxt/framework/issues/2122